From: Colin Ian King <colin.i.king@gmail.com>
Subject: mm: truncate: remove redundant initialization of new_order
Date: Thu, 6 Apr 2023 10:58:17 +0100

Variable new_order is being initialized with a value that is not read, the
variable is being re-assigned later.  Remove the initialization.

Link: https://lkml.kernel.org/r/20230406095817.703426-1-colin.i.king@gmail.com
Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/truncate.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/truncate.c~mm-truncate-remove-redundant-initialization-of-new_order
+++ a/mm/truncate.c
@@ -213,7 +213,7 @@ bool truncate_inode_partial_folio(struct
 {
 	loff_t pos = folio_pos(folio);
 	unsigned int offset, length, remaining;
-	unsigned int new_order = folio_order(folio);
+	unsigned int new_order;
 
 	if (pos < start)
 		offset = start - pos;
_
